lib/cn/kvs_mblk_desc.c

```diff
@@ -71,38 +75,36 @@ mblk_munmap(struct mpool *mp, struct kvs_mblk_desc *md)
 }
 
 merr_t
-mblk_madvise(const struct kvs_mblk_desc *md, size_t off, size_t len, int advice)
+mblk_madvise_pages(const struct kvs_mblk_desc *md, size_t pg, size_t pg_cnt, int advice)
 {
-    int rc;
-    const size_t wlen = md->wlen_pages * PAGE_SIZE;
+    const size_t wlen_pages = md->wlen_pages;
+    size_t ra_pages;
+    size_t chunk = 0;
 
-    assert(md->map_base);
+    if (pg > wlen_pages)
+        return merr(EINVAL);
 
-    if (len == 0)
+    if (pg + pg_cnt > wlen_pages)
+        pg_cnt = wlen_pages - pg;
+
+    if (pg_cnt == 0)
```

Comment:
Seems like (pg_cnt == 0) should the first check in this function?

Reply:
Then I'd have to check it twice, right?  This way I can do it just once.  Lemme look again..

Reply:
If you change the EINVAL check to (pg >= wlen_pages) then pg_cnt can never be clamped to zero at line 88.

Reply:
Good point!  I guess if someone calls starting at the end then they are probably broken and off in the weeds...